src/main/java/com/bp3x/raidbot/commands/util/Event.java

```diff
@@ -1,59 +1,171 @@
+
+    public ArrayList<Member> getTentativePlayers() { return tentativePlayers; }
+
+    public String getEventId() { return eventId; }
+
+    public LocalDateTime getTime() { return time; }
+
+
+    public enum EventPlayerStatus {
+        ACCEPTED,
+        DECLINED,
+        TENTATIVE
+    }
+
+    /**
+     * Load event information from event config json.
+     *
+     * @param shortName - event shortname to load from.
+     */
+    public void load(String shortName) throws RaidBotRuntimeException {
+        try {
+            JsonElement element = JsonParser.parseReader(new FileReader("event.json"));
+            JsonObject eventConfig = element.getAsJsonObject();
+            // need an additional object to hold the specific event and its values
+            JsonObject eventAsJSON = eventConfig.getAsJsonObject(shortName);
+
+            this.longName = RaidBotUtils.getValueFromJSON(LONG_NAME_KEY, eventAsJSON);
+            this.playerCount = Integer.parseInt(RaidBotUtils.getValueFromJSON(PLAYER_COUNT_KEY, eventAsJSON));
+
+        } catch (RuntimeException rte) {
+            throw new RaidBotRuntimeException("There was an error parsing the event.json file. Shutting down the bot.");
+
+        } catch (FileNotFoundException e) {
+            throw new RaidBotRuntimeException("Unable to find event json file. Shutting down the bot.");
+        }
+    }
+
+    /**
+     * Adds the event to the list of currently planned events
+     * @param message The Discord message (embed post) associated with the event
+     */
+    public void registerEvent(Message message) {
+        plannedEventsList.put(this, message);
+    }
+
+    /**
+     * Registers a user's availability for the event, removing existing status if present
+     * @param player - The user to RSVP for the event
+     * @param status - The status to set the player as for the event
+     */
+    public void setPlayerStatus(Member player, EventPlayerStatus status) {
+        acceptedPlayers.remove(player);
+        declinedPlayers.remove(player);
+        tentativePlayers.remove(player);
+
+        switch(status) {
+            case ACCEPTED:
+                acceptedPlayers.add(player);
+                break;
+            case DECLINED:
+                declinedPlayers.add(player);
+                break;
+            case TENTATIVE:
+                tentativePlayers.add(player);
+                break;
+        }
+    }
+
+    /**
+     * Checks whether a given event is currently defined in event config json.
+     * @param shortName - event shortname to test existence for
+     */
+    public static boolean eventExists(String shortName) {
+        try {
+            JsonElement element = JsonParser.parseReader(new FileReader("event.json"));
+            JsonObject eventConfig = element.getAsJsonObject();
+
+            return eventConfig.has(shortName);
+        } catch (FileNotFoundException e) {
+            log.error("Unable to find event json file.", e);

```

Comment:
Should we throw a RaidBotRuntimeException here to shut down the bot?

Reply:
Yeah, that makes sense. I'll add it.